### PR TITLE
#162727469 Create articles

### PR DESCRIPTION
**Pull request by wombolo**
#### What does this PR do?
This enables authorised users create new articles that can be read by other users

#### Description of Task to be completed?
- Users are able to CREATE articles and READ articles created by other Users.

- Users are able to ONLY UPDATE and DELETE articles that i have authored.

- Given two different users create articles of the same title, the app should generate unique slugs such that the articles don't create conflict.

#### How should this be manually tested?
After migrations and installations of dependencies,

To Create an article, run POST > `http://localhost:9001/api/v1/arts` VALUES:
```
title:Cirrhosis of the Sky,
description:Cirrhosis of the Sky, The Syntax Podcasts isn't suitable for all ages,
categoryId:1,
media:[{"url":"https://farm3.staticflickr.com/2817/33968464326_a6f9cbc754_k","extension":"jpg"},   {"url":"https://images.pexels.com/photos/1047540/pexels-photo-1047540.jpeg?dl&fit=crop&crop=entropy&w=1280&h=853","extension":"jpeg"}]
```

To Edit an article, run PUT > `http://localhost:9001/api/v1/arts` VALUES:
```
title:Cirrhosis of the Sky,
description:Cirrhosis of the Sky, The Syntax Podcasts isn't suitable for all ages,
categoryId:1,
media:[{"url":"https://farm3.staticflickr.com/2817/33968464326_a6f9cbc754_k","extension":"jpg"},   {"url":"https://images.pexels.com/photos/1047540/pexels-photo-1047540.jpeg?dl&fit=crop&crop=entropy&w=1280&h=853","extension":"jpeg"}]
```

To DELETE an article, run PUT > `http://localhost:9001/api/v1/arts/article-slug`

#### Any background context you want to provide?

- At least a Category is needed to create articles

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162727469

#### Screenshots (if appropriate)
image

#### Questions:

### CODE REVIEW

**Review by abejide001 — CHANGES_REQUESTED** [review]
Fix the houndci errors

**Comment by julietezekwe** [comment]
Kindly check why the test is dropping. Check the catch blocks and the EmailNotificationAPI file.

**Comment by daniellamarr** [comment]
Take all comments into consideration, then resolve conflicts

**Review by daniellamarr — COMMENTED** [review]
Resolve conflicts

**Comment by wombolo** [comment]
> Kindly check why the test is dropping. Check the catch blocks and the EmailNotificationAPI file.

Coverage rose on the EmailNotificationAPI file, that catch blocks appear to be the issue

File: src/routes/artsRoute.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+import ArtController from '../controllers/ArtsController';
+import { TokenAuthenticate } from '../helpers/index';
+
+const authRouter = express.Router();
```
Comment (daniellamarr) [inline src/routes/artsRoute.js]: Your file name and functionalities do not correlate, please attend to this

File: src/routes/artsRoute.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+authRouter.put('/:slug', TokenAuthenticate.tokenVerify, ArtController.update);
+
+authRouter.delete(
+  '/:slug',
+  TokenAuthenticate.tokenVerify,
+  ArtController.delete
+);
+
+export default authRouter;
```
Comment (daniellamarr) [inline src/routes/artsRoute.js]: Same thing here, you are using `authRouter` function inside `artsRoute.js`

**Comment by julietezekwe** [comment]
Please can you check why test is dropping

**Review by julietezekwe — COMMENTED** [review]
I dropped a comment on test coverage

File: src/controllers/ArtsController.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
+import models from '../db/models';
+import { TokenAuthenticate, Response, Slugify } from '../helpers/index';
```
[the remark this answers is not in the document]
Comment (abejide001) [reply src/controllers/ArtsController.js]: fix this errror

[the remark this answers is not in the document]
Comment (wombolo) [reply src/controllers/ArtsController.js]: fixed

**Review by abejide001 — APPROVED** [review]

**Review by Proception — APPROVED** [review]

**Review by andela-dbamidele — APPROVED** [review]

File: test/controllers/artsControllerTest.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+          expect(res.body.messages).eql('Validation Errors Occurred');
+          expect(res.status).to.equal(400);
+          expect(res.body.status).eql('Not Ok');
+          done(err);
+        });
+    });
+
+    it('it should not update an article with invalid slug', (done) => {
+      chai.request(app)
+        .put(`/api/v1/arts/ss-slug`)
```
Comment (andela-dbamidele) [inline test/controllers/artsControllerTest.js]: Please use single quotes and not template strings

Comment (wombolo) [reply test/controllers/artsControllerTest.js]: Thanks, I'm working on it.

**Pull request merged**